src/auth.rs

```diff
@@ -841,24 +831,18 @@ impl ConfiguredOidc {
             .get_http_request_header("cookie")
             .ok_or(PluginError::SessionCookieNotFoundError)?;
 
-        // Split cookie by ; and filter for the cookie name.
-        let cookies = cookie
-            .split(';')
-            .filter(|x| x.contains(self.plugin_config.cookie_name.as_str()))
-            .filter(|x| !x.contains(format!("{}-nonce", self.plugin_config.cookie_name).as_str()));
-
-        // Check if cookies have values
-        for cookie in cookies.clone() {
-            if cookie.split('=').collect::<Vec<&str>>().len() < 2 {
-                return Err(PluginError::SessionCookieNotFoundError);
-            }
-        }
-
-        // Then split all cookies by = and get the second element before joining all values together.
-        let values = cookies
-            .map(|x| x.split('=').collect::<Vec<&str>>()[1])
-            .collect::<Vec<&str>>()
-            // Join the cookie values together again.
+        let cookie_name = &self.plugin_config.cookie_name;
+        let num_parts: u8 = self
+            .get_cookie(&format!("{cookie_name}-parts"))
+            .unwrap_or_default()
+            .parse()
+            .map_err(|_| PluginError::SessionCookieNotFoundError)?;
```

Comment:
Technically its not the Session Cookie right? We should add another Error for this type as Nonce also has its own error

Reply:
I don't think this is worth the effort. I would consider all of the cookies together "the session cookie" because it doesn't really make a difference to the user which is missing, the end result is the same. In fact, I would like to refactor the error handling completely to use `anyhow` instead of defining a custom error type with `thiserror`. Initially I thought we might `match` on the error variant somewhere and do different things depending on the error, but we don't actually do that. Furthermore, `PluginError` is just a catch-all error type anyway, that doesn't reflect which errors a function can actually throw. `fn validate_cookie(&self) -> Result<AuthorizationState, PluginError>` will never return `Err(PluginError::CodeNotFoundInCallbackError)`, but the return type makes it look like it could.